pyfilesync/tests/console.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Testing module console.
+"""
+
+from __future__ import unicode_literals
+import unittest
+import sys
+from io import StringIO
+import pyfilesync
+from pyfilesync.console import main
+
+
+# This is tested class. Can have too many method
+class TestConsoleModule(unittest.TestCase):  # pylint: disable=R0904
+    """ Module testsCase. """
+
+    def setUp(self):
+        """ Setup environment. """
```

Comment:
Unittest, `SetUp` wywołuje się przed każdym testem, `tearDown` po zakończeniu. W tych metodach mniej więcej zapisujemy sobie standardowe strumienia wyjścia i błędów do pamięci, a na ich miejsce dodajemy `StringIO` gdzie będą wrzucane outputy (czyli trafią do stringa) które normalnie idą na ekran. W `tearDown` przywracam `stdout` i `stderr` do starej wersji.

Reply:
Czy możesz coś wiecej o tym powiedzieć? W jakim celu zapisujesz do Stringa standardowe strumenia wyjścia i błędy? To tak jak w linuxie rzucasz w NULL? 

Swoją drogą fajny zapis testów.

Reply:
Te strumienie zapisuje żeby wszystko co wypiszesz printem trafiło do zmiennej self.tmp_stdout. Wtedy możesz sobie ogarnąć czy to co wypisuje program jest poprawne. Gdyby tego nie było to ten tekst wypisałby się po prostu w programie, a ty nie miałbyś jak go sprawdzić

Reply:
Nie do końca bo tam jest jeszcze przypisanie do self.tmp_stdout gdzie potem sobie w jednym teście to sprawdzasz. Więc tą informację masz w polu obiektu. Tak czy inaczej to masz rację w tym, że ten tekst z printów się nie wypisze na ekranie, więc okno nie będzie zaśmiecane listingiem samego programu konsolowego.